### PR TITLE
Registries

### PR DESCRIPTION
**Pull request by pbadillatorrealba**
This pre adds a new feature to dashai: registers.

Following the idea of implementing a plugin structure, the registries are classes that store and manage the avialable plugins for DashAI.

The registries were implemented in such a way that:
- They are general: that is, they do not depend on a specific component/module, instead, the component is a parameter of the register. This implies that from inheritance, we can very easily implement the registers for any type of component while not replicating code.
- Non-task plugins are able to set a relation (mapping) with the tasks (through a mixin).
- The mapping mixin also have methods that, given a task or a component, return its compatibles component or tasks respectively.

All the core functionality of the registers is tested.

### CODE REVIEW

File: DashAI/back/main.py [hunk 1]
```diff
@@ -1,16 +1,47 @@
...
+)
+from DashAI.back.models import (
+    SVC,
+    KNeighborsClassifier,
+    RandomForestClassifier,
+)
+from DashAI.back.models.classes.getters import (
+    filter_by_parent,
+    get_model_params_from_task,
+)
```
Comment (cristian-tamblay) [inline DashAI/back/main.py]: Unused, i think we can remove from ignore this file on pre-commit check, so it throws errors on unused imports again.

File: DashAI/back/main.py [hunk 2]
```diff
@@ -1,16 +1,47 @@
+# isort: skip_file
+import json
```
Comment (cristian-tamblay) [inline DashAI/back/main.py]: Unused

File: DashAI/back/models/base_model.py [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 from DashAI.back.config_object import ConfigObject
 
 
-# version == 1.0.0
-class Model(ConfigObject):
+class BaseModel(ConfigObject, metaclass=ABCMeta):
     """
     Abstract class of all machine learning models.
-    The models must implement the save and load methods.
+    All models must extend this clas and implement save and load methods.
```
Comment (cristian-tamblay) [inline DashAI/back/models/base_model.py]: typo

File: DashAI/back/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from DashAI.back.models.base_model import BaseModel
+from DashAI.back.models.scikit_learn.k_neighbors_classifier import KNeighborsClassifier
+from DashAI.back.models.scikit_learn.random_forest_classifier import (
+    RandomForestClassifier,
+)
+from DashAI.back.models.scikit_learn.svc import SVC
```
Comment (cristian-tamblay) [inline DashAI/back/models/__init__.py]: Why we import on init?

File: DashAI/back/models/classes/numerical_wrapper_for_text.py [hunk 5]
```diff
@@ -2,11 +2,11 @@
 
 from sklearn.feature_extraction.text import CountVectorizer
 
-from DashAI.back.models.classes.sklearn_like_model import SklearnLikeModel
-from DashAI.back.models.classes.text_classification_model import TextClassificationModel
+from DashAI.back.models.scikit_learn.sklearn_like_model import SklearnLikeModel
```
Comment (cristian-tamblay) [inline DashAI/back/models/classes/numerical_wrapper_for_text.py]: Why we don't import this on DashAI/back/models/__init__.py?

File: DashAI/back/tasks/translation_task.py [hunk 6]
```diff
@@ -3,30 +3,19 @@
...
-    @staticmethod
-    def create():
-        task = TranslationTask()
-        return task
-
-    def validate_dataset(self, dataset: DatasetDict, class_column: Union[str, int]):
-        """
-        TODO: Implement this validation
-        """
-        return None
```
Comment (cristian-tamblay) [inline DashAI/back/tasks/translation_task.py:28]: Why remove this? Didn't we agree that tasks validate their datasets? Even though this method is not yet implemented

File: DashAI/back/tasks/base_task.py [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+            parameters = execution.get_params()
+            # executionBytes = execution.save()
+
+            self.experimentResults[execution.MODEL] = {
+                "train_results": trainResults,
+                "test_results": testResults,
+                "parameters": parameters,
+                # " executionBytes": executionBytes,
+            }
+
```
Comment (cristian-tamblay) [inline DashAI/back/tasks/base_task.py:145]: I think we should remove these, it shouldnt be the task's job to run executions and parse parameters

File: DashAI/back/tasks/translation_task.py [hunk 8]
```diff
@@ -3,30 +3,19 @@
...
+class TranslationTask(BaseTask):
     """
     Abstract class for translation tasks.
     Here you can change the methods provided by class TranslationTask.
     """
 
-    NAME: str = "TranslationTask"
+    name: str = "TranslationTask"
     SOURCE: str = ""
     TARGET: str = ""
```
Comment (cristian-tamblay) [inline DashAI/back/tasks/translation_task.py]:
```
source: str = ""
target: str = ""
```

File: DashAI/back/main.py (hunk 1 above)
Comment (pbadillatorrealba) [reply DashAI/back/main.py]: Ok

File: DashAI/back/main.py (hunk 2 above)
Comment (pbadillatorrealba) [reply DashAI/back/main.py]: Deleted.

File: DashAI/back/models/classes/numerical_wrapper_for_text.py (hunk 5 above)
Comment (pbadillatorrealba) [reply DashAI/back/models/classes/numerical_wrapper_for_text.py]: Ok

**Review by cristian-tamblay — APPROVED** [review]

**Pull request merged**